ballerina/tests/jms_connection_tests.bal

```diff
@@ -0,0 +1,150 @@
+
+@test:Config {
+    groups: ["connection"]
+}
+isolated function testCreateConnectionSuccess() returns error? {
+    Connection connection = check new (
+        initialContextFactory = "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl = "tcp://localhost:61616"
+    );
+    check connection->close();
+}
+
+@test:Config {
+    groups: ["connection"]
+}
+isolated function testCreateConnectionInvalidInitialContextFactory() returns error? {
+    Connection|Error connection = new (
+        initialContextFactory = "io.sample.SampleMQInitialContextFactory",
+        providerUrl = "tcp://localhost:61616"
+    );
+    test:assertTrue(connection is Error, 
+        "Connection created with invalid initial context factory");
+    if connection is Error {
+        test:assertEquals(connection.message(), 
+            "Error occurred while connecting to broker: Cannot instantiate class: io.sample.SampleMQInitialContextFactory", 
+            "Invalid connection init error message");
+    }
+}
+
+@test:Config {
+    groups: ["connection"]
+}
+isolated function testCreateConnectionInvalidProviderUrl() returns error? {
+    Connection|Error connection = new (
+        initialContextFactory = "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl = "tcp://localhost:61615"
+    );
+    test:assertTrue(connection is Error, 
+        "Connection created with invalid provider URL");
+    if connection is Error {
+        test:assertEquals(connection.message(), 
+            "Error occurred while connecting to broker: Could not connect to broker URL: tcp://localhost:61615. Reason: java.net.ConnectException: Connection refused (Connection refused)", 
+            "Invalid connection init error message");
+    }
+}
+
+// @test:Config {
+//     groups: ["connection"]
+// }
+isolated function testCreateConnectionInvalidCredentials() returns error? {
+    Connection|Error connection = new (
+        initialContextFactory = "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl = "tcp://localhost:61616",
+        username = "testuser",
+        password = "testpassword"
+    );
+    test:assertTrue(connection is Error, 
+        "Connection created with invalid credentials");
+    if connection is Error {
+        test:assertEquals(connection.message(), 
+            "Error occurred while connecting to broker: Could not connect to broker URL: tcp://localhost:61615. Reason: java.net.ConnectException: Connection refused (Connection refused)", 
+            "Invalid connection init error message");
+    }
+}
+
+@test:Config {
+    groups: ["connection"]
+}
+isolated function testConnectionRestart() returns error? {
+    Connection connection = check new (
+        initialContextFactory = "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl = "tcp://localhost:61616"
+    );
+    check connection->stop();
+    runtime:sleep(2);
+    check connection->'start();
```

Comment:
is there a way to confirm whether the `'start`/`stop` are successfull?

Reply:
For both `start` and `close` there will be an error whenever the relevant action was unsuccessful. So, if there is an error the test case will fail.

Reply:
ack